base/binaryplatforms.jl

```diff
@@ -186,7 +186,7 @@ end
 function Base.show(io::IO, p::Platform)
     str = string(platform_name(p), " ", arch(p))
     # Add on all the other tags not covered by os/arch:
-    other_tags = sort(collect(filter(kv -> kv[1] ∉ ("os", "arch"), tags(p))))
+    other_tags = sort(filter(kv -> kv[1] ∉ ("os", "arch"), tags(p))))
```

Comment:
This could use sort!, no?

Reply:
It is an iterator, see: https://github.com/JuliaLang/julia/pull/50119#discussion_r1224517302

Reply:
Right, `filter` produces a vector here, not an iterable.

Reply:
I think `tags(p)` produces a `Dict`. The original code produced a sorted vector of pairs. This new code produces an OrderedDict. I believe this would work efficiently
```
-other_tags  = sort(collect(filter(kv -> kv[1] ∉ ("os", "arch"), tags(p))))
+other_tags  = sort(kv for kv in tags(p) if kv[1] ∉ ("os", "arch"))
```
though @sjkelly's original change is also correct:
```
-other_tags  = sort(collect(filter(kv -> kv[1] ∉ ("os", "arch"), tags(p))))
+other_tags  = sort!(collect(filter(kv -> kv[1] ∉ ("os", "arch"), tags(p))))
```